modules/gateways/Blockonomics/Blockonomics.php

```diff
@@ -327,39 +384,241 @@ public function insertOrderToDb($id_order, $address, $value, $bits) {
+	public function isOrderPending($orders) {
+		foreach ($orders as $order) {
+			//check if status 0 or 1
+			if($order->status == 0 || $order->status == 1){
+				return $order;
+			}
+		}
+		return;
+	}
+
+	/*
+	 * Check for unused address for the currency linked to the order
+	 */		
+	public function isOrderWaiting($orders, $blockonomics_currency) {
+		foreach ($orders as $order) {
+			//check for currency address already waiting
+			if($order->blockonomics_currency == $blockonomics_currency && $order->status == -1){
+				//Check if existing order is expired
+				$current_time = time();
+				$total_time = $this->getTimePeriod() * 60;
+				$clock = $order->timestamp + $total_time - $current_time;
+				if($clock < 0){
+					$order->bits = $this->getNewAmount($order->value, $order->blockonomics_currency);
+					$order->timestamp = $current_time;
+				}
+				$this->updateOrderExpected($order->addr, $order->blockonomics_currency, $order->timestamp, $order->bits);
+				return $order;
+			}
+		}
+		return;
 	}
 
 	/*
-	 * Try to get order row from db by uuid
+	 * Check for unused address or create new
+	 */	
+	public function isOrderNew($orders, $blockonomics_currency) {
+		foreach ($orders as $order) {
+			//check for new blank order in db
+			if($order->addr == ""){
+				$new_addresss_response = $this->getNewAddress($blockonomics_currency);
+				if ($new_addresss_response->response_code == 200){
+					$order->addr = $new_addresss_response->address;
+				}else{
+					exit($new_addresss_response->message);
+				}
+
+				$this->updateOrderAddress($order->id_order, $order->addr, $blockonomics_currency);
+				$order->blockonomics_currency = $blockonomics_currency;
+				$order->bits = $this->getNewAmount($order->value, $order->blockonomics_currency);
+				$order->timestamp = time();
+				$this->updateOrderExpected($order->addr, $order->blockonomics_currency, $order->timestamp, $order->bits);
+				return $order;
+			}
+		}
+		// blank order already used, create a new blank order with same uuid
+		// fix this
+		$order_uuid = $this->newOrder($orders[0]->value, $orders[0]->id_order);
+		$order = $this->getOrderByHash($order_uuid, $blockonomics_currency);
+		return $order;
+	}
+
+	/*
+	 * Get existing or create new order
+	 */	
+	public function getOrderByHash($order_uuid, $blockonomics_currency) {
+		// Fetch all orders by uuid
+		$all_orders_by_uuid = $this->getAllOrdersByHash($order_uuid);
+		if(!$all_orders_by_uuid){
+			return;
+		}
+		// Check for pending payments
+		$pending_payment = $this->isOrderPending($all_orders_by_uuid);
+		if($pending_payment){
+			return $pending_payment;
+		}
+		// Check for existing address
+		$address_waiting = $this->isOrderWaiting($all_orders_by_uuid, $blockonomics_currency);
+		if($address_waiting){
+			$address_waiting->currency = getCurrency(getClientsDetails()['user_id'])['code'];
+			return $address_waiting;
+		}
+		// Check for new order
+		$new_order = $this->isOrderNew($all_orders_by_uuid, $blockonomics_currency);
```

Comment:
Pass isOrderNew a single order as input that already has the correct $blockonomics_currency

Reply:
Depends upon
> Make the getAllOrdersbyHash return a dict